src/Raven.Client/ServerWide/Operations/UniqueValueOperation.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using Raven.Client.Documents.Conventions;
+using Raven.Client.Documents.Session;
+using Raven.Client.Http;
+using Raven.Client.Json;
+using Raven.Client.Util;
+using Sparrow.Json;
+
+namespace Raven.Client.ServerWide.Operations
+{
+    public class GetRawUniqueValueResult
+    {
+        public BlittableJsonReaderObject Value;
+        public long Index;
+    }
+
+    public class GetUniqueValueResult<T>
+    {
+        public T Value;
+        public long Index;
+    }
+
+    public class GetUniqueValue<T> : IServerOperation<GetUniqueValueResult<T>>
+    {
+        private readonly string _key;
+
+        public GetUniqueValue(string key)
+        {
+            _key = key;
+        }
+
+        public RavenCommand<GetUniqueValueResult<T>> GetCommand(DocumentConventions conventions, JsonOperationContext context)
+        {
+            return new GetUniqueValueCommand(_key, conventions);
+        }
+
+        private class GetUniqueValueCommand : RavenCommand<GetUniqueValueResult<T>>
+        {
+            private readonly string _key;
+            private readonly DocumentConventions _conventions;
+            
+            public GetUniqueValueCommand(string key, DocumentConventions conventions = null)
+            {
+                if (string.IsNullOrEmpty(key))
+                    throw new ArgumentNullException(nameof(key), "The key argument must have value");
+
+                _key = key;
+                _conventions = conventions ?? DocumentConventions.Default;
+                ResponseType = RavenCommandResponseType.Raw;
+            }
+
+            public override bool IsReadRequest { get; }
+
+            public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, ServerNode node, out string url)
+            {
+                url = $"{node.Url}/unique?key={_key}";
+                var request = new HttpRequestMessage
+                {
+                    Method = HttpMethods.Get,
+                };
+                return request;
+            }
+
+            public override void SetResponseRaw(HttpResponseMessage response, Stream stream, JsonOperationContext context)
```

Comment:
Why do we need to use `SetReeponseRaw`, why not let the ReuqestExecuter do the parsing for us?